scripts/src/test/org/oppia/android/scripts/RegexPatternValidationCheckTest.kt

```diff
@@ -0,0 +1,81 @@
+package org.oppia.android.scripts
+
+import java.io.File
+import kotlin.test.assertEquals
+import org.oppia.android.scripts.RegexPatternValidationCheck
+import org.junit.Test
+import org.junit.Rule
+import org.junit.Before
+import org.junit.rules.TemporaryFolder
+import org.junit.rules.ExpectedException
+import org.oppia.android.scripts.ScriptResultConstants
+
+class RegexPatternValidationCheckTest {
+
+  @Rule
+  @JvmField
+  public var tempFolder: TemporaryFolder = TemporaryFolder()
+
+  @Rule
+  @JvmField
+  var thrown: ExpectedException = ExpectedException.none()
+
+  @Before
+  fun initTestFilesDirectory() {
+    val testFilesDirectory = tempFolder.newFolder("testfiles")
+  }
+
+  @Test
+  fun testFileNamePattern_validFileNamePattern_fileNamePatternIsCorrect() {
+    val appLayerMimic = tempFolder.newFolder("testfiles", "app", "src", "main")
+    val prohibitedFileNamePattern = tempFolder.newFile(
+      "testfiles/app/src/main/TestActivity.kt"
+    )
+
+    runScript(tempFolder.getRoot().toString() + "/testfiles", arrayOf("app"))
```

Comment:
We should have a way to verify the console output from the script. Could we maybe redirect the system standard/error outputs to an in-memory stream so that we can validate the console output? This will make nicer tests since it lets us verify that a pass is actually a pass, that we have multiple failures when there are multiple failures, and that the output is correctly formatted.

Please also add tests to verify output format & multiple errors can be properly thrown.

Reply:
done.
The tests now verify the console output too for every pass and failure in addition to check the result status of the script.
Also, a test is added for multiple failures. PTAL.